prow/plugins/project/project.go

```diff
@@ -0,0 +1,257 @@
+	GetProjectColumns(projectID int) ([]github.ProjectColumn, error)
+	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
+	GetProjectCard(projectID int, number int) (*github.ProjectCard, error)
+	MoveProjectCard(projectCardID int, newColumnID int) error
+	DeleteProjectCard(projectCardID int) error
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	msgForTeam := func(team plugins.ProjectConfig) string {
+		return fmt.Sprintf(projectTeamMsg, team.ProjectMaintainersTeam, team.ProjectMaintainersID)
+	}
+
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The project plugin allows members of a configurable GitHub team to set the project and column on an issue or pull request.",
+		Config: func(repos []string) map[string]string {
+			configMap := make(map[string]string)
+			for _, repo := range repos {
+				team, exists := config.ProjectConfigMap[repo]
+				if exists {
+					configMap[repo] = msgForTeam(team)
+				}
+			}
+			configMap[""] = msgForTeam(config.ProjectConfigMap[""])
+			return configMap
+		}(enabledRepos),
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/project <board>, /project <board> <column>, or /project clear <board>",
+		Description: "Updates the project board and column for an issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Members of the project maintainers GitHub team can use the '/project' command.",
+		Examples:    []string{"/project 0.5.0", "/project 0.5.0 To do", "/project clear 0.4.0"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, &e, pc.PluginConfig.ProjectConfigMap)
+}
+
+func updateProjectNameToIDMap(projects []github.Project) {
+	for _, project := range projects {
+		projectNameToIDMap[project.Name] = project.ID
+	}
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, projectConfigMap map[string]plugins.ProjectConfig) error {
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	matches := projectRegex.FindStringSubmatch(e.Body)
+	if len(matches) != 2 && len(matches) != 3 {
+		return fmt.Errorf(invalidNumArgs)
+	}
+
+	proposedProject := matches[1]
+	proposedColumnName := ""
+	if len(matches) > 2 && proposedProject != clearKeyword {
+		proposedColumnName = matches[2]
+	}
+	projectToClear := ""
+	if proposedProject == clearKeyword {
+		projectToClear = matches[2]
+	}
+
+	// See if the current user is part of the team in order to set the project/column
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	projectMaintainers, err := gc.ListTeamMembers(projectConfigMap[repo].ProjectMaintainersID, github.RoleAll)
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, person := range projectMaintainers {
```

Comment:
I think we're doing this in a number of plugins now -- can we get a helper method on the github client that wraps this up?

```go
func (c *GitHubClient) TeamContainsMember(teamID, memberLogin string) bool
```